phpserver/router.php

```diff
@@ -28,74 +28,93 @@
         $val = json_encode($val);
     }
 
-    echo 'debug: ' . $val . PHP_EOL . '<br/>' . PHP_EOL;
+    error_log('debug: '.$val);
 };
 
 /**
- * Note: the code below is experimental and not intended to be used outside development environment.
- * The code is protected against running outside of PHP built-in web server.
+ * Caution, this is very experimental stuff
+ * no guarantee for working result
+ * has tons of potential big security holes
  */
 
 if (php_sapi_name() === 'cli-server') {
-    $debug($_SERVER["REQUEST_URI"]);
+
+    $debug("URI: {$_SERVER["REQUEST_URI"]}");
     if (preg_match('/^\/(index|get|static)\.php(\/)?/', $_SERVER["REQUEST_URI"])) {
         return false;    // serve the requested resource as-is.
     }
 
     $path = pathinfo($_SERVER["SCRIPT_FILENAME"]);
     $url   = pathinfo(substr($_SERVER["REQUEST_URI"], 1));
     $route = parse_url(substr($_SERVER["REQUEST_URI"], 1))["path"];
-
-    $debug($path);
-    $debug($route);
+    $ext = @pathinfo($route)['extension'];
```

Comment:
are there cases when pathinfo is throwing an error, or is it because the array access for 'extension' is sometimes not working?